setup.py

```diff
@@ -34,6 +34,7 @@
     "delta-spark>=2.2.0",
     "pandas<2.0.0",  # From 2.0.0 onwards, pandas does not support iteritems() anymore, spark.createDataFrame will fail
     "numpy<1.24",  # From 1.24 onwards, module 'numpy' has no attribute 'bool'.
+    "more_itertools",
```

Comment:
Create LPP ticket for this, otherwise re-implement a single function. Don't add whole library for the sake of a function